core/src/main/java/org/bptlab/cepta/operators/EnrichDelayWithCoordinatesFunction.java

```diff
@@ -0,0 +1,97 @@
+package org.bptlab.cepta.operators;
+
+import com.mongodb.reactivestreams.client.MongoClient;
+import com.mongodb.reactivestreams.client.MongoCollection;
+import com.mongodb.reactivestreams.client.MongoDatabase;
+import org.apache.flink.api.common.functions.RichFlatMapFunction;
+import org.apache.flink.streaming.api.functions.async.ResultFuture;
+import org.apache.flink.streaming.api.functions.async.RichAsyncFunction;
+import org.apache.flink.util.Collector;
+import org.bptlab.cepta.config.MongoConfig;
+import org.bptlab.cepta.models.internal.notifications.notification.NotificationOuterClass;
+import org.bptlab.cepta.models.internal.types.coordinate.CoordinateOuterClass;
+import org.bptlab.cepta.utils.database.Mongo;
+import org.bptlab.cepta.utils.database.mongohelper.SubscriberHelpers;
+import org.bptlab.cepta.utils.notification.NotificationHelper;
+import org.bson.Document;
+
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.List;
+
+
+public class EnrichDelayWithCoordinatesFunction extends RichFlatMapFunction<NotificationOuterClass.Notification, NotificationOuterClass.Notification> {
+    /* This Function takes an inputstream of DelayNotifications and enriches the events
+       with information about the coordinations of the dedicated station */
+
+    public EnrichDelayWithCoordinatesFunction(MongoConfig mongoConfig){
+        this.mongoConfig = mongoConfig;
+        this.tableName = "eletastations";
+    }
+
+    public EnrichDelayWithCoordinatesFunction(String tableName, MongoConfig mongoConfig){
+        this.mongoConfig = mongoConfig;
+        this.tableName = tableName;
+    }
+
+    private Hashtable<String, CoordinateOuterClass.Coordinate> coordinateMapping =  new Hashtable<>();
+
+    public Hashtable<String, CoordinateOuterClass.Coordinate> getMapping(){ return coordinateMapping;}
+
+    private MongoConfig mongoConfig = new MongoConfig();
+    private transient MongoClient mongoClient;
+    private String tableName;
+
+    private boolean readInStationData(){
+        MongoDatabase database = mongoClient.getDatabase("replay"/*mongoConfig.getName()*/);
```

Comment:
maybe we should also make the database a member variable? to be able to specifiy which db hosts the "base data"